lib/net/nforceif/include/lwipopts.h

```diff
@@ -332,6 +362,25 @@
  */
 #define LWIP_HAVE_LOOPIF                0
 
+/*
+   ------------------------------------
+   ---------- Thread options ----------
+   ------------------------------------
+*/
+
+/**
+ * TCPIP_THREAD_STACKSIZE: The stack size used by the main tcpip thread.
+ * The stack size value itself is platform-dependent, but is passed to
+ * sys_thread_new() when the thread is created.
+ */
+#define TCPIP_THREAD_STACKSIZE          4096
+
+/**
+ * DEFAULT_THREAD_STACKSIZE: The stack size used by any other lwIP thread.
+ * The stack size value itself is platform-dependent, but is passed to
+ * sys_thread_new() when the thread is created.
+ */
+#define DEFAULT_THREAD_STACKSIZE        4096
```

Comment:
nit: How are these thread stacksizes used?

I'm always a bit paranoid about some code doing something like `DEFAULT_THREAD_STACKSIZE+4 /* make room for return address */` (fantasy example - not real code in lwip) and then it consumes more than one page.

Also is one page enough? Could there be some bad recursion or something like `alloca` for large packet streams?

Reply:
In our case they're only used for the tcp thread, which takes packets in pbufs from the driver via an mbox, does some parsing and endianess conversion, and then hands them to application threads via another mbox.

All buffers are managed by pbufs, so I don't expect any large stack usage, and it never caused issues during testing. With `LWIP_TCPIP_CORE_LOCKING_INPUT`, the tcp thread should barely be doing anything, anyway.